torch/csrc/jit/codegen/cuda/executor.h

```diff
@@ -0,0 +1,69 @@
+#pragma once
+#include <torch/csrc/jit/codegen/cuda/executor_launch_params.h>
+#include <torch/csrc/jit/codegen/cuda/executor_utils.h>
+#include <torch/csrc/jit/codegen/cuda/fusion.h>
+#include <torch/csrc/jit/codegen/cuda/ir_all_nodes.h>
+#include <torch/csrc/jit/codegen/cuda/ir_cloner.h>
+#include <torch/csrc/jit/codegen/cuda/ir_printer.h>
+#include <torch/csrc/jit/codegen/cuda/kernel.h>
+#include <torch/csrc/jit/codegen/cuda/lower2device.h>
+
+namespace torch {
+namespace jit {
+namespace fuser {
+namespace cuda {
+
+struct TORCH_CUDA_API CompileOptions {
+  int device = 0;
+};
+
+class TORCH_CUDA_API FusionExecutor {
```

Comment:
add comment describing what a FusionExecutor is/does?

Reply:
A FusionExecutor seems to be usable for multiple compilations / runs, right? At the same time, it's stateful. What is the rationale for this approach (as opposed to a more "functional" design - compile(fusion) -> kernel. kernel object can be executed)